tsdb/index/index.go

```diff
@@ -1251,15 +1247,12 @@ type Range struct {
 // for all postings lists.
 func (r *Reader) PostingsRanges() (map[labels.Label]Range, error) {
```

Comment:
`PostingsRanges` is never called?

Reply:
Apperently it's not. I can see it called in one Mimir's test though.

I feel we can remove it, also the `Range` type which is only used here, but I'd rather do that in a separate PR instead, leaving this one only for optimization changes.